File: src/values/i32.rs
pub struct ValueI32 {
    value: Option<i32>,
}

impl ValueI32 {
    pub fn init_empty() -> ValueI32 {
        return ValueI32 { value: None };
    }
    
    pub fn init_with(value: i32) -> ValueI32 {
        return ValueI32 { value: Some(value) };
    }
    
    pub fn show(&self, end: char) {
        if let Some(val) = self.value {
            print!("{val}{end}");
        } else {
            print!("null{end}");
        }
    }
    
    pub fn is_null(&self) -> bool {
        return self.value.is_none();
    }
    
    pub fn get_type_representation(&self) -> &str {
        return "i32";
    }
    
    pub fn get_value(&self) -> Option<i32> {
        return self.value;
    }
}

File: src/values/i32.cpp
#include "values/i32.hpp"
using namespace common::values;

ValueI32::ValueI32()
    : value(std::nullopt) {
    
}

ValueI32::ValueI32(int32_t value)
    : value(value) {
    
}

void ValueI32::show(char end) const {
    if (this->value)
        std::cout << "i32<" << this->value.value() << '>' << end;
    else
        std::cout << "i32<null>" << end;
}

bool ValueI32::isNull() const {
    return !this->value.has_value();
}

std::string ValueI32::getTypeRepr() const {
    return "i32";
}

ValueKind ValueI32::getKind() const {
    return ValueKind::I32;
}

std::optional<int32_t> ValueI32::getValue() const {
    return this->value;
}